src/plugins/visualize/public/application/legacy_app.js

```diff
@@ -160,10 +160,19 @@ export function initVisualizeApp(app, deps) {
               );
             }
 
+            // This delay is needed to prevent some navigation issues in Firefox/Safari.
+            // see https://github.com/elastic/kibana/issues/65161
+            const delay = res => {
+              return new Promise(resolve => {
+                setTimeout(() => resolve(res), 0);
+              });
+            };
+
             return data.indexPatterns
               .ensureDefaultIndexPattern(history)
               .then(() => savedVisualizations.get($route.current.params))
               .then(getResolvedResults(deps))
+              .then(delay)
               .catch(
```

Comment:
Won't this `.catch` no longer catch errors from the `getResolvedResults` function, since the delay function can never throw an error? This looks like it could inadvertently cause our error handling behavior to be different